### PR TITLE
Fix 0.6 abstract type declaration depwarn

### PR DESCRIPTION
**Pull request by yuyichao**
~~**DO NOT MERGE NOW!!!!**~~

~~This requires https://github.com/JuliaLang/Compat.jl/pull/325 and a new tag. I'm using this (and coming) PR's to check if the implementation is correct.~~

Ready now.

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,2 +1,2 @@
 julia 0.5
-Compat 0.9.5
+Compat 0.16.3
```
Comment (tkelman) [inline REQUIRE]: new features should increment minor

Comment (yuyichao) [inline REQUIRE]: K. That seems to be a new policy after 0.9.5?

Comment (tkelman) [inline REQUIRE]: I've been starting to tell people to use semver, yes

**Comment by yuyichao** [comment]
Compat tagged and CI reenabled.

**Comment by yuyichao** [comment]
Ping.

**Pull request merged**